firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -201,6 +203,17 @@ public void onClose(Status status) {
                 handleWriteStreamClose(status);
               }
             });
+
+    networkReachabilityMonitor.onNetworkReachabilityChange(
```

Comment:
I think this is fine. It's pretty different from what iOS does, but my understanding is that the Android implementation has significantly less control over gRPC, so trying to bring this in line with iOS is probably impractical. One difference is that iOS does an error stop while this implementation performs a graceful stop. Graceful stop resets backoff, which is better for reestablishing connection, though it would result in more unsuccessful attempts to connect in case of losing connection.